packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -121,11 +121,21 @@ export interface SharedTreeBranchEvents<TEditor extends ChangeFamilyEditor, TCha
 	dispose(): void;
 }
 
+/**
+ * Events related to branch trimming. This is done as a performance optimization.
+ */
+export interface BranchTrimmingEvents {
+	/**
+	 * Fired when commits are trimmed from the branch.
+	 */
+	branchTrimmed(trimmedRevisions: RevisionTag[]): void;
```

Comment:
IMO something like `ancestryTrimmed` rather than `branchTrimmed` would give some more immediate information about what this event does without being too many more characters - up to you though.